pkg/ottl/parser.go

```diff
@@ -480,6 +481,35 @@ func (p *Parser[K]) ParseValueExpression(raw string) (*ValueExpression[K], error
 	}
 
 	return &ValueExpression[K]{
-		getter: getter,
+		getter: &StandardGetSetter[K]{
+			Getter: func(ctx context.Context, tCtx K) (any, error) {
+				val, err := getter.Get(ctx, tCtx)
+				if err != nil {
+					return nil, err
+				}
+				switch v := val.(type) {
+				case map[string]any:
+					m := pcommon.NewMap()
+					if err := m.FromRaw(v); err != nil {
+						return nil, err
+					}
+					return m, nil
+				case []any:
+					for index, elem := range v {
+						// make sure also nested maps within a slice are returned in their raw form
+						if m, ok := elem.(pcommon.Map); ok {
+							v[index] = m.AsRaw()
+						}
+					}
+					s := pcommon.NewSlice()
```

Comment:
I guess returning an `pcommon.Slice` here would be a breaking change, since the original behavior was returning `[]any`. I'm wondering if we should address the slices inconsistencies here or in a separate PR, maybe by changing the `listGetter` for example.

Reply:
Alright, I have reverted the change for the slices again to keep returning `[]any` for now. If needed we can do the adaptation of the listGetter in a separate PR